staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -182,20 +187,37 @@ func (s *EtcdOptions) ApplyTo(c *server.Config) error {
 		return nil
 	}
 
-	s.addEtcdHealthEndpoint(c)
+	if err := s.addEtcdHealthEndpoint(c); err != nil {
+		return err
+	}
 	c.RESTOptionsGetter = &SimpleRestOptionsFactory{Options: *s}
 	return nil
 }
 
 func (s *EtcdOptions) ApplyWithStorageFactoryTo(factory serverstorage.StorageFactory, c *server.Config) error {
-	s.addEtcdHealthEndpoint(c)
+	if err := s.addEtcdHealthEndpoint(c); err != nil {
+		return err
+	}
 	c.RESTOptionsGetter = &storageFactoryRestOptionsFactory{Options: *s, StorageFactory: factory}
 	return nil
 }
 
-func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) {
+type ETCDEndpointsGetter func() []string
+
+func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
+	endpointsGetter, err := factory.Endpoints(s.StorageConfig)
```

Comment:
this can likely be simplified once https://github.com/kubernetes/kubernetes/pull/65027 merges

Reply:
sure